### PR TITLE
MAM-3619-make-report-userpost-menu-items-red-add-confirmation-alert

### PR DESCRIPTION
**Pull request by bnolens**
[MAM-3619 : Make 'Report user/post' menu items red; add confirmation alert](https://linear.app/theblvd/issue/MAM-3619/make-report-userpost-menu-items-red-add-confirmation-alert)

### CODE REVIEW

File: Mammoth/Views/Cells/PostCardCell/PostCardCell.swift [hunk 1]
```diff
@@ -927,15 +927,29 @@ extension PostCardCell {
 
 // MARK: - Context menu creators
 extension PostCardCell {
-    private func createContextMenuAction(_ title: String, _ buttonType: PostCardButtonType, isActive: Bool, onPress: @escaping PostCardButtonCallback) -> UIAction {
+    private func createContextMenuAction(_ title: String, _ buttonType: PostCardButtonType, isActive: Bool, onPress: @escaping PostCardButtonCallback, attributes:  UIMenuElement.Attributes = []) -> UIAction {
+
```
Comment (rileyhCode) [inline Mammoth/Views/Cells/PostCardCell/PostCardCell.swift:931]: Consider consolidating these three createContextMenuAction() into a single global function.

**Review by rileyhCode — APPROVED** [review]
One suggestion to consider — not sure if worth doing.
Then, feel free to merge.

File: Mammoth/Views/Cells/PostCardCell/PostCardCell.swift (hunk 1 above)
Comment (bnolens) [reply Mammoth/Views/Cells/PostCardCell/PostCardCell.swift:931]: This requires a little bit of a refactoring of the code to do this correctly. We currently have 3 different createContextMenuAction methods. The onPress behavior specifically is where the differences exist. If you think this is important to clean up I'd do this as a separate ticket.

Comment (rileyhCode) [reply Mammoth/Views/Cells/PostCardCell/PostCardCell.swift:931]: Gotcha. No need to do it now then.

**Pull request merged**